SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -3184,3 +3204,39 @@ void USpatialNetDriver::PopCrossServerRPCSender(AActor* SenderActor)
 	check(GSenderStack.Last() == SenderActor);
 	GSenderStack.Pop();
 }
+
+void USpatialNetDriver::OnLevelAddedToWorld(ULevel* Level, UWorld* InWorld)
+{
+	Super::OnLevelAddedToWorld(Level, InWorld);
+
+	SCOPE_CYCLE_COUNTER(STAT_OnLevelAddedToWorld);
+
+	// When a sublevel gets reloaded soon after it was unloaded, the entities in that level might still be in the view.
+	// In that case, we need to tell the ActorSystem to restore channels/references for those actors.
+	if (!IsServer() && Level != nullptr && ActorSystem.IsValid())
+	{
+		FUnrealObjectRef LevelObjectRef = FUnrealObjectRef::GetStablyNamedObjectRef(Level);
+
+		for (AActor* Actor : Level->Actors)
+		{
+			if (Actor == nullptr || !Actor->IsNetStartupActor())
+			{
+				continue;
+			}
+
+			FUnrealObjectRef ActorRef;
+			// Usually actors in a level will have the level as their outer
+			if (Actor->GetOuter() == Level)
+			{
+				FString ActorName = UWorld::RemovePIEPrefix(Actor->GetFName().ToString());
+				ActorRef = FUnrealObjectRef(0, 0, ActorName, LevelObjectRef, true);
+			}
+			else
+			{
+				ActorRef = FUnrealObjectRef::GetStablyNamedObjectRef(Actor);
+			}
+
+			ActorSystem->RestoreStablyNamedActor(ActorRef, Actor);
+		}
+	}
```

Comment:
I'm concerned that this is going to be slow for large sublevels since we're copying a lot of strings when creating the object refs.
The main alternative I'm considering is instead of doing this on level load, only check & restore the channel/refs when receiving an update or component add for an entity that doesn't have an actor channel.

Reply:
Yeah this seems a valid concern. Doing it here enables it to be hotfixed onto 12.1 more easily though. Perhaps we could run a test in-game and see if it produces noticeable hitches?